### PR TITLE
Public API Changes for Optimize

### PR DESCRIPTION
**Pull request by ashumadobe**
Updated adobe-journey-optimizer-decisioning sdk docs with new public APIs for UpdatePropositions with completion callback

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/pages/edge/adobe-journey-optimizer-decisioning/api-reference.md [hunk 1]
```diff
@@ -104,13 +104,33 @@ iOS
 
 <Tabs query="platform=ios&api=update-propositions"/>
 
+## updatePropositions
```
Comment (spoorthipujariadobe) [inline src/pages/edge/adobe-journey-optimizer-decisioning/api-reference.md:107]: Can we call this updatePropositionsWithCompletionHandler to differentiate it from the other API? Here's an example from Media SDK have we have similar set of APIs
https://developer.adobe.com/client-sdks/solution/adobe-media-analytics/api-reference/#createtracker
https://developer.adobe.com/client-sdks/solution/adobe-media-analytics/api-reference/#createtrackerwithconfig

File: src/pages/home/base/mobile-core/tabs/api-reference.md [hunk 2]
```diff
@@ -1398,7 +1398,8 @@ public interface AdobeCallback<T> {
 
 The `AdobeCallbackWithError` class provides the interface to receive results or an error when the asynchronous APIs perform the requested action.
 
-When using this class, if the request cannot be completed within the default timeout or an unexpected error occurs, the request is stopped and the fail method is called with the corresponding `AdobeError`.
+When using this class, if the request cannot be completed within the default timeout or an unexpected error occurs, the request is stopped and the fail method is called with the corresponding`
+`.
```
Comment (spoorthipujariadobe) [inline src/pages/home/base/mobile-core/tabs/api-reference.md:1402]: Did we accidentally omit `AdobeError` after corresponding?

**Review by calebk1 — COMMENTED** [review]
Couple of comments!

**Review by spoorthipujariadobe — APPROVED** [review]

File: src/pages/edge/adobe-journey-optimizer-decisioning/api-reference.md [hunk 3]
```diff
@@ -104,13 +104,32 @@ iOS
 
 <Tabs query="platform=ios&api=update-propositions"/>
 
+## updatePropositionsWithCompletionHandler
+
+This API dispatches an Event for the Edge network extension to fetch decision propositions, for the provided decision scopes array, from the decisioning services enabled in the Experience Edge. The returned decision propositions are cached in-memory in the Optimize SDK extension and can be retrieved using `getPropositions` API.
```
[the remark this answers is not in the document]
Comment (ashumadobe) [reply src/pages/edge/adobe-journey-optimizer-decisioning/api-reference.md]: done

File: src/pages/edge/adobe-journey-optimizer-decisioning/api-reference.md [hunk 4]
```diff
@@ -104,13 +104,32 @@ iOS
...
+
+<InlineAlert variant="help" slots="text"/>
+
+Completion callback passed to `updatePropositions` supports network timeout and fatal errors returned by edge network along with fetched propositions data. SDK's internal retry mechanism handles the recoverable HTTP errors therefore recoverable HTTP errors are not returned through this callback.
```
[the remark this answers is not in the document]
Comment (ashumadobe) [reply src/pages/edge/adobe-journey-optimizer-decisioning/api-reference.md]: done

**Review by calebk1 — APPROVED** [review]
LGTM!

**Review by spoorthipujariadobe — COMMENTED** [review]
LGTM! Missing one small update on this page to change `score` from `Int` to `Double` in Offer iOS class
https://developer.adobe.com/client-sdks/edge/adobe-journey-optimizer-decisioning/api-reference/#offer

**Pull request merged**